docs/source/killed.rst

```diff
@@ -0,0 +1,105 @@
+which also died. After a configurable number of deaths, Dask decides to blame the
+task itself, and returns this exception.
+
+In every case, the first place to look for further information is the logs of
+the given worker, which may well give a complete description of what happened. These
+logs are printed by the worker to it's "standard output", which may be the text
+console from which you launched the worker, or some logging system maintained by
+the cluster infrastructure.
+
+In all cases, the scheduler will notice that the worker has gone, either because
+of an explicit de-registration, or because the worker no longer produces heartbeats,
+and it should be possible to reroute tasks to other workers and have the system
+keep running.
+
+Worker chose to exit
+''''''''''''''''''''
+
+Workers may exit in normal functioning because they have been asked to, e.g.,
+they received a keyboard interrupt (^C), or the scheduler scaled down the cluster.
+In such cases, the work that was being done by the worker will be redirected to
+other workers, if there are any left.
+
+You should expect to see the following message at the end of the worker's log:
+
+.. pull-quote::
+
+   distributed.dask_worker - INFO - End worker
+
+In these cases, there is not normally anything which you need to do, since the
+behaviour is expected.
+
+Unrecoverable Exception
+'''''''''''''''''''''''
+
+The worker is a python process, and like any other code, an exception may occur
+which causes the process to exit. One typical example of this might be a
+version mismatch between the packages of the client and worker, so that
+a message sent to the worker errors while being unpacked. There are a number of
+packages that need to match, not only ``dask`` and ``distributed``.
+
+In this case, you should expect to see the full python traceback in the worker's
+log. In the event of a version mismatch, this might be complaining about a bad
+import or missing attribute. However, other fatal exceptions are also possible,
+such as trying to allocate more memory than the system has available, or writing
+temporary files without appropriate permissions.
+
+To assure that you have matching versions, you should run (more recent versions
+of distributed may do this automatically)
+
+.. code-block::
+
+   client.get_versions(check=True)
+
+For other errors, you might want to run the computation in your local client, if
+possible, or try grabbing just the task that errored and using
+:func:`recreate_error_locally <distributed.recreate_exceptions.ReplayExceptionClient.recreate_error_locally>`,
+as you would for ordinary exceptions happening during task execution.
+
+Specifically for connectivity problems (e.g., timeout exceptions in the worker
+logs), you will need to diagnose your networking infrastructure, which is more
+complicated than can be described here. Commonly, it may involve logging into
+the machine running the affected worker
+(although you can :ref:`ipylaunch`).
```

Comment:
I like all of the situations that you've brought up here (versions, running out of memory, network connectivity.   I think that it might make sense to enumerate these into their own small subsections and then include them in the TOC.  I think that this might also make it easier to extend this document in the future as we remember more ways in which workers die.  It will be easier, I think, to extend this list, than to add other cases into the narrative flow that you have here (although your flow reads more nicely than a list).